File: tests/java/TestDecorator.java
public class TestDecorator {
  public static void main(String[] args) {
    System.out.println("Testing Java Decorator...");
    try {
      Client.main(args);
    } catch (Exception e) {
      e.printStackTrace();
      System.exit(1);
    }
    System.out.println("Java Decorator Test Passed!");
  }
}

File: tests/c/TestDecorator.c
#define main pattern_main
#include "../../oop_in_c/Decorator/Decorator.c"
#undef main
#include <stdio.h>

int main() {
  printf("Testing C Decorator...\n");
  // Call pattern main
  pattern_main();
  printf("C Decorator Test Passed!\n");
  return 0;
}
